extensions/adobe/experience/campaign/experienceevent.example.click.json

```diff
@@ -0,0 +1,100 @@
+{
+  "xdm:timestamp": 1522321544448,
+  "xdm:metrics": {
+    "https://ns.adobe.com/xdm/data/metrics/web/link-clicks": {
+      "xdm:value": 1
+    }
+  },
+  "xdm:environment": {
+    "xdm:type": "browser",
+    "xdm:browserDetails": {
+      "xdm:name": "Chrome"
+    },
+    "xdm:operatingSystem": "MAC OS"
+  },
+  "xdm:device": {
+    "xdm:typeID": "IMac",
+    "xdm:typeIDService": "https://ns.adobe.com/xdm/external/proprietary"
```

Comment:
This example looks unrealistic.

Reply:
@trieloff This is derived from an actual event that currently Adobe Campaign receives when someone clicks on a web-link inside an email. Can you help me understand what ways this should be improved?

Regarding the `typeIDService`, at Adobe Campaign, we parse the user-agent string and have our own enumeration of values that are set as the value of `device` depending upon some regex rules. This might look bad and we have plans to move to DeviceAtlas. But until this happens, we have data that flows using this. 
I assume there may be similar producers and this "proprietary" enum was created to honor a contract between the producers and consumers that the service is not a standard one but their own internal service. Basically, an enum for "others". Yes, this would not work when the consumer doesn't know the producer or consumes from multiple producers using different "proprietary" service.

I first thought of extending the device schema (say `campaign/device.schema`) to add this "proprietary" enum value but I couldn't find a way to extend an enum set through an extended schema. Also, `context/experienceEvent` has the `device` field. I would want this to be overridden to `campaign\device` in `campaign/experienceEvent`. I don't think we can do that unless we create a fresh field.
As a simple solution, I could just change this to call it `CampaignDeviceService` if that seems best but I didn't want to include Adobe specific things in the core xdm.
Any suggestions?

Reply:
I would just give this a URL that includes the name `Adobe Campaign`. @asteed might have a suggestion that is consistent with our schemas.

Just having a bucket for "others" is not useful, but knowing that "Adobe Campaign" though this is an IMac, adds a lot of value.